pallets/runtime/src/bridge.rs

```diff
@@ -0,0 +1,294 @@
+use frame_system::{self as system, ensure_root, ensure_signed};
+use polymesh_primitives::{traits::IdentityCurrency, AccountKey, IdentityId, Signatory};
+use polymesh_runtime_balances as balances;
+use polymesh_runtime_common::{traits::CommonTrait, Context};
+use polymesh_runtime_identity as identity;
+use sp_core::H256;
+use sp_std::collections::btree_map::BTreeMap;
+use sp_std::{convert::TryFrom, prelude::*};
+
+pub trait Trait: multisig::Trait {
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+    type Proposal: From<Call<Self>> + Into<<Self as identity::Trait>::Proposal>;
+}
+
+/// The intended recipient of POLY exchanged from the locked ERC20 tokens.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum IssueRecipient<AccountId> {
+    Account(AccountId),
+    Identity(IdentityId),
+}
+
+/// A unique lock-and-mint bridge transaction containing Ethereum transaction data and a bridge nonce.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BridgeTx<AccountId, Balance> {
+    /// Bridge signer runtime nonce.
+    pub nonce: u64,
+    /// Recipient of POLY on Polymesh: the deposit address or identity.
+    pub recipient: IssueRecipient<AccountId>,
+    /// Amount of tokens locked on Ethereum.
+    pub amount: Balance,
+    /// Ethereum token lock transaction hash.
+    pub tx_hash: H256,
+}
+
+/// A transaction that is pending a valid identity KYC.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PendingTx<AccountId, Balance> {
+    /// The identity on which the KYC is pending.
+    pub did: IdentityId,
+    /// The pending transaction.
+    pub bridge_tx: BridgeTx<AccountId, Balance>,
+}
+
+/// Either a pending transaction or a `None` or an error.
+type IssueResult<T> = sp_std::result::Result<
+    Option<PendingTx<<T as frame_system::Trait>::AccountId, <T as CommonTrait>::Balance>>,
+    DispatchError,
+>;
+
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        /// The bridge relayer set address is not set.
+        RelayersNotSet,
+        /// The signer does not have an identity.
+        IdentityMissing,
+        /// Failure to credit the recipient account.
+        CannotCreditAccount,
+        /// Failure to credit the recipient identity.
+        CannotCreditIdentity,
+        /// The origin is not the relayer set multisig.
+        BadCaller,
+        /// The recipient DID has no valid KYC.
+        NoValidKyc,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Bridge {
+        /// The multisig account of the bridge relayer set. The genesis signers must accept their
+        /// authorizations to be able to get their proposals delivered.
+        Relayers get(relayers) build(|config: &GenesisConfig| {
+            if config.signatures_required > u64::try_from(config.signers.len()).unwrap_or_default()
+            {
+                panic!("too many signatures required");
```

Comment:
Instead of panic shouldn't we spit the error?

Reply:
This is a startup error. By stopping execution and emitting the error message we prompt the user to fix the genesis config and restart. Besides, the closure doesn't seem to be able to return a `Result`.